### PR TITLE
docs: Suggest mathematically correct use of sum_over_time

### PR DESCRIPTION
**Pull request by samjewell**
See the docs change itself in the commit.

### CODE REVIEW

File: docs/querying/functions.md [hunk 1]
```diff
@@ -669,6 +669,8 @@ over time and return an instant vector with per-series aggregation results:
 * `min_over_time(range-vector)`: the minimum value of all points in the specified interval.
 * `max_over_time(range-vector)`: the maximum value of all points in the specified interval.
 * `sum_over_time(range-vector)`: the sum of all values in the specified interval.
+   Note: for a mathematically correct [numerical integration](https://en.wikipedia.org/wiki/Numerical_integration) using the [rectangle method](https://en.wikipedia.org/wiki/Riemann_sum) also multiply by a factor to expand the rectangles to take up the full time range without gaps.
+   eg. for a `_per_second` gauge metric, with time-period between samples (scrape interval) of 15s, multiply by 15: `sum_over_time(<metric_prefix>_per_second[<duration>]) * 15`
```
Comment (samjewell) [inline docs/querying/functions.md:673]: When using `sum_over_time` recently I was trying to find the increase log GB ingested over a day. 
The metric that we were already working with was  `…_logs_…_bytes_received_per_second` - ie a gauge metric representing a rate. So I needed to find the area under the curve - hence why I used `sum_over_time`.

But I'm realising that, if I had access to a counter metric instead (something like `…_bytes_received_total`, then I could use the `increase()` function instead. This seems preferable, as when using `increase()` you don't need to know the scrape-interval. And missing samples won't throw the results off too.

So shall I include something along the lines of "Prefer using `increase()` with a counter metric to integrating a rate metric, as it's more robust to missing samples and changes in scrape interval". WDYT?

**Comment by beorn7** [comment]
:thinking: This paragraph is a really compressed, dealing with all the `..._over_time` functions in one go. Expanding one of the paragraphs a lot to explain usage implications and best practices seems out of place. Furthermore, your notes aren't even sufficient. If you really wanted to explain "mathematically correct" integration, you also needed to explain the problem with changing sampling intervals, missing samples, and series starting or stopping (see the long winded discussion over in #14440).

We are running into the general problem here that (most of) the documentation on prometheus.io is in fact terse reference documentation, and that we don't have a good place yet for documentation that is more oriented towards explaining usage. We need to address this problem, but sprinkling explanation all over in between the reference documentation will make things even worse.

I think what we have here is actually beautiful because it is so short and it says on the tin exactly what you get. How to use counters vs. already `rate`'d metrics and how to navigate the pitfalls of the `..._over_time` function could be part of the "Best Practices" section or maybe a tutorial in the "Tutorial" section. (Don't ask we what the precise difference is between "Tutorials" and "Guides" … another sign that we really need to properly structure the website.)

**Comment by juliusv** [comment]
> Don't ask we what the precise difference is between "Tutorials" and "Guides" … another sign that we really need to properly structure the website.

I think it was originally meant by the people who introduced those sections as differentiated on https://rasa.com/blog/writing-a-technical-tutorial-or-guide-to-support-the-open-source-community/:

> ### What is the difference between a tutorial and a guide?
> **A tutorial** teaches the audience a new idea or concept by showing how to accomplish a goal. Tutorials take you through a technical process step-by-step and tend to get straight to the point.
>
> **A guide** is usually focused on achieving a particular task and can assume some level of prerequisite knowledge. Like tutorials, they have a clear focus but also include cautionary advice and additional information. Guides may also suggest multiple approaches to accomplishing the task.

**Comment by beorn7** [comment]
Off-topic, of course, for this PR, but still: Even if we have a good definition of the difference, the question is if we should really have separate sections. Including the subtly different https://prometheus.io/docs/guides/go-application/ vs. https://prometheus.io/docs/tutorials/instrumenting_http_server_in_go/ …

**Comment by beorn7** [comment]
To come back to the topic here: I guess that means we could have a _guide_ around the topic "how to use `..._over_time` functions" or maybe even specifically "How to use `sum_over_time` and how to avoid it (by using the underlying counter)". I could also see this as an article in the "Best Practices" section, though.

**Comment by samjewell** [comment]
> In any case, such an article would be a way bigger effort than what's proposed in this PR. @samjewell you are more than welcome to write such an article, but no obligations, of course. But I think it makes most sense to close this PR (which doesn't prevent reusing the content here for such an article).

Got it. Thanks for the consideration and feedback.

**Pull request closed without merging**